src/components/OpenLayersMap.js

```diff
@@ -0,0 +1,60 @@
+import  { useEffect, useRef } from 'react';
+import 'ol/ol.css';
+import { Map, View } from 'ol';
+import { Tile as TileLayer, Vector as VectorLayer } from 'ol/layer';
+import { OSM, Vector as VectorSource } from 'ol/source';
+import { Style, Icon } from 'ol/style';
+import Feature from 'ol/Feature';
+import Point from 'ol/geom/Point';
+import LocationPin from '../images/marker-icon.png';
+
+const OpenLayersMap = ({ onLocationSelected }) => {
```

Comment:
Why do you have this Map component as well?

Reply:
i need one map component for the form and one for the main page!